base/bin/build-init

```diff
@@ -20,25 +28,29 @@ if [[ "${REMOTE_CODEBASE_METHOD}" == "rsync" ]]; then
 else
 	# Checkout sources on the remote host
 	echo "Checking out codebase via git..."
-	build-exec "git clone --branch="${GIT_BRANCH_NAME}" --depth 50 ${GIT_REPO_URL} . && git reset --hard ${GIT_COMMIT_HASH} && ls -la"
+	if [[ "${REMOTE_BUILD_DIR_CLEANUP}" == 1 ]]; then
+		build-exec "git clone --branch="${GIT_BRANCH_NAME}" --depth 50 ${GIT_REPO_URL} . && git reset --hard ${GIT_COMMIT_HASH} && ls -la"
+	else
+		build-exec "(git fetch origin && git reset --hard origin/$GIT_BRANCH_NAME && ls -la) || (git clone --branch="${GIT_BRANCH_NAME}" --depth 50 ${GIT_REPO_URL} . && git reset --hard ${GIT_COMMIT_HASH} && ls -la)"
```

Comment:
I'd split this into two steps: check if the remote build dir is an existing repo, then fetch or clone based on that.
This will be readable better and also won't throw errors for `git fetch` upon the first run.

Reply:
Please also add comments explaining the logic and actions that follow.

Reply:
Added logic to keep environment cleanup and code deployment separated. Comments are in the code explaining the logic around this.